TRENZ.Lib.RazorMail.System.Net/SystemNetMailClient.cs

```diff
@@ -0,0 +1,89 @@
+﻿using System.Net;
+using System.Net.Mail;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+using JetBrains.Annotations;
+
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+using TRENZ.Lib.RazorMail.Extensions;
+using TRENZ.Lib.RazorMail.Models;
+using TRENZ.Lib.RazorMail.Services;
+
+using RazorMailMessage = TRENZ.Lib.RazorMail.Models.MailMessage;
+using SystemNetMailMessage = System.Net.Mail.MailMessage;
+
+namespace TRENZ.Lib.RazorMail;
+
+public class SystemNetMailClient(IOptions<SmtpAccount> accountOptions, ILogger<SystemNetMailClient> logger) : BaseMailClient(accountOptions)
+{
+
+    [MustDisposeResource]
+    private SmtpClient CreateClient()
+    {
+        var client = new SmtpClient(Account.Host);
+
+        client.Port = Account.Port;
+        client.EnableSsl = Account.TLS;
+        client.Credentials = new NetworkCredential(Account.Login, Account.Password);
+
+        return client;
+    }
+
+    /// <inheritdoc />
+    protected override async Task SendInternalAsync(RazorMailMessage message, CancellationToken cancellationToken)
+    {
+        using var nativeMessage = ConvertToNative(message);
+        using var client = CreateClient();
+
+        cancellationToken.ThrowIfCancellationRequested();
+
+        logger.LogInformation("Sending mail from {From} to {Recipients} (CC: {Cc}, BCC: {Bcc}) with subject {Subject}",
+            nativeMessage.From, nativeMessage.To, nativeMessage.CC, nativeMessage.Bcc, nativeMessage.Subject);
+
+        await client.SendAsync(nativeMessage);
+    }
+
+    [MustDisposeResource]
+    private static SystemNetMailMessage ConvertToNative(RazorMailMessage razorMessage)
```

Comment:
maybe instead make this an extension method as well (`razorMessage.ToMailMessage()`)?

Reply:
I moved all conversion methods to extension classes